ui/pages/confirmations/components/confirm/footer/useIsDangerButton.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import useSignatureSecurityAlertResponse from '../../../hooks/useSignatureSecurityAlertResponse';
+import { currentConfirmationSelector } from '../../../selectors';
+import { SecurityAlertResponse } from '../../../types/confirm';
+import { BlockaidResultType } from '../../../../../../shared/constants/security-provider';
+
+const useIsDangerButton = () => {
+  const [isDangerButton, setIsDangerButton] = useState(false);
+  const currentConfirmation = useSelector(currentConfirmationSelector);
+
+  const currentSecurityAlertId = (
+    currentConfirmation?.securityAlertResponse as SecurityAlertResponse
```

Comment:
As mentioned in the other comment, is this `securityAlertResponse` property different to the one below, why do we query again by ID?

Reply:
thread above https://github.com/MetaMask/metamask-extension/pull/24402#discussion_r1595361446